### PR TITLE
feat: support schema for key and web

### PR DESCRIPTION
**Pull request by tipusinghaw**
#### What
- Supported schema for key and web
- Insufficient balance implementation
- Implement a single endpoint for schema creation for all type of schemas

### CODE REVIEW

File: apps/ledger/src/schema/schema.service.ts [hunk 1]
```diff
@@ -36,207 +38,215 @@ export class SchemaService extends BaseService {
...
+                schemaDetails
+              );
+  
+              const attributesArray = JSON.parse((await saveResponse).attributes);
+              (await saveResponse).attributes = attributesArray;
+              delete (await saveResponse).lastChangedBy;
+              delete (await saveResponse).lastChangedDateTime;
+              return saveResponse;
+  
+            } else if ('finished' === responseObj.state) {
```
Comment (GHkrishna) [inline apps/ledger/src/schema/schema.service.ts:208]: Can we take the states inside of an Enum?

File: apps/ledger/src/schema/schema.service.ts [hunk 2]
```diff
@@ -36,207 +38,215 @@ export class SchemaService extends BaseService {
...
-            );
-
-            const attributesArray = JSON.parse((await saveResponse).attributes);
-            (await saveResponse).attributes = attributesArray;
-            delete (await saveResponse).lastChangedBy;
-            delete (await saveResponse).lastChangedDateTime;
-            return saveResponse;
-
+  
+            if ('finished' === responseObj.schema.state) {
```
Comment (GHkrishna) [inline apps/ledger/src/schema/schema.service.ts:189]: Can we take the states inside of an Enum?

File: apps/ledger/src/schema/schema.service.ts [hunk 3]
```diff
@@ -36,207 +38,215 @@ export class SchemaService extends BaseService {
...
+            const agentDetails = await this.schemaRepository.getAgentDetailsByOrgId(orgId);
+            if (!agentDetails) {
+              throw new NotFoundException(
+                ResponseMessages.schema.error.agentDetailsNotFound,
+                { cause: new Error(), description: ResponseMessages.errorMessages.notFound }
+              );
+            }
+            const { agentEndPoint, orgDid } = agentDetails;
+            const getAgentDetails = await this.schemaRepository.getAgentType(orgId);
+            // eslint-disable-next-line yoda
```
Comment (GHkrishna) [inline apps/ledger/src/schema/schema.service.ts:126]: Yoda disabled

**Review by KulkarniShashank — APPROVED** [review]
LG

**Review by GHkrishna — APPROVED** [review]
LG

**Pull request merged**